scipy/stats/_correlation.py

```diff
@@ -0,0 +1,200 @@
+    # √n·ξn(X, Y) → N(0, 2/5) in distribution as n → ∞"
+    if y_continuous:  # [1] Theorem 2.1
+        return np.sqrt(2 / 5) / np.sqrt(n)
+
+    # "Suppose that X and Y are independent. Then √n·ξn(X, Y)
+    # converges to N(0, τ²) in distribution as n → ∞
+    # [1] Eq. 2.2 and surrounding math
+    i = np.arange(1, n + 1)
+    u = np.sort(r, axis=-1)
+    v = np.cumsum(u, axis=-1)
+    an = 1 / n**4 * np.sum((2*n - 2*i + 1) * u**2, axis=-1)
+    bn = 1 / n**5 * np.sum((v + (n - i)*u)**2, axis=-1)
+    cn = 1 / n**3 * np.sum((2*n - 2*i + 1) * u, axis=-1)
+    dn = 1 / n**3 * np.sum((l * (n - l)), axis=-1)
+    tau = (an - 2*bn + cn**2) / dn**2
+
+    return np.sqrt(tau) / np.sqrt(n)
+
+
+def _xi_correlation_iv(y_continuous, method):
+    # Input validation for `xi_correlation`
+    # x, y, `axis` input validation taken care of by decorator
+
+    if y_continuous not in {True, False}:
+        raise ValueError('`y_continuous` must be boolean.')
+
+    if not isinstance(method, stats.PermutationMethod):
+        method = method.lower()
+        message = "`method` must be 'asymptotic' or a `PermutationMethod` instance."
+        if method != 'asymptotic':
+            raise ValueError(message)
+
+    return y_continuous, method
+
+
+def _unpack(res):
+    return res.statistic, res.pvalue
+
+
+@_axis_nan_policy_factory(SignificanceResult, paired=True, n_samples=2,
+                          result_to_tuple=_unpack, n_outputs=2, too_small=1)
+def xi_correlation(x, y, *, axis=0, y_continuous=False, method='asymptotic'):
+    r"""Compute the xi correlation and perform a test of independence
+
+    The xi correlation coefficient is a measure of association between two
+    variables; the value tends to be close to zero when the variables are
+    independent and close to 1 when there is a strong association. Unlike
+    other correlation coefficients, the xi correlation is effective even
+    when the association is not monotonic.
+
+    Parameters
+    ----------
+    x, y : array-like
+        The samples: corresponding observations of the independent and
+        dependent variable. The (N-d) arrays must be broadcastable.
+    axis : int, default: 0
+        Axis along which to perform the test.
+    method : 'asymptotic' or `PermutationMethod` instance, optional
+        Selects the method used to calculate the *p*-value.
+        Default is 'asymptotic'. The following options are available.
+
+        * ``'asymptotic'``: compares the standardized test statistic
+          against the normal distribution.
+        * `PermutationMethod` instance. In this case, the p-value
+          is computed using `permutation_test` with the provided
+          configuration options and other appropriate settings.
+
+    y_continuous : bool, default: False
+        Whether `y` is assumed to be drawn from a continuous distribution.
+        If `y` is drawn from a continuous distribution, results are valid
```

Comment:
My understanding from the paper is that if Y is continuous, the (asymptotic) standard error has a closed form, and if Y is not continuous, the standard error depends on the distribution of Y which is in turn estimated (consistently) from the input Y data.

For large sample size, consistency should be all that is needed. For moderate sample size, the closed form formula might be more efficient (in statistical sense). For small sample, the asymptotic distribution might not be so accurate anyway, so a little more noise in the standard error probably doesn’t matter.

Reply:
Was there anything to change here, or just confirming understanding?